runhouse/resources/hardware/on_demand_cluster.py

```diff
@@ -464,6 +467,38 @@ def teardown(self):
         Example:
             >>> rh.ondemand_cluster("rh-cpu").teardown()
         """
+        # TODO [SB]: remove the den_auth check once we will get status of clusters without den_auth as well.
+        if self.den_auth:
+            try:
+                # cluster_config = self.config()
+                cluster_status_data = self.status()
+                status_data = {
+                    "status": "terminated",
+                    "resource_type": cluster_status_data.get("cluster_config").get(
```

Comment:
do we have access to the cluster config separately here? meaning it seems like its embedded inside the cluster status data, is it the exact same data?

Reply:
It mostly the same data. The cluster_config embedded inside the cluster status contains the following additional keys: domain and server_host. 

The only reason we need to access the cluster_config is to get the resource_type, but we can extract it using `self.__class__.__base__.__name__ `(for sub-resources of cluster) or` self.__class__.__name__` (for clusters)